plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/grok/GrokProcessor.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.processor.grok;
+
+import org.elasticsearch.ingest.Data;
+import org.elasticsearch.ingest.processor.Processor;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Properties;
+
+public final class GrokProcessor implements Processor {
+
+    public static final String TYPE = "grok";
+    public static final String PATTERNS_PATH = "/org/elasticsearch/ingest/processor/grok";
+    public static final String PATTERN_PROPERTIES = "pattern.properties";
+
+    private final String matchField;
+    private final String matchPattern;
+    private final Grok grok;
+
+    public GrokProcessor(String matchField, String matchPattern) throws IOException {
+        this.matchField = matchField;
+        this.matchPattern = matchPattern;
+
+        this.grok = new Grok();
+
+        // TODO(talevy): finalize a nicer way to manage these pattern files (properties file or filepaths)
```

Comment:
ok, I think I got it. I followed what `jvm-example` does to manage its `config` files. An additional assemblies task is executed in maven to load the config files into `config/ingest`